client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -713,15 +713,25 @@ private TopicPublishInfo tryToFindTopicPublishInfo(final String topic) {
         TopicPublishInfo topicPublishInfo = this.topicPublishInfoTable.get(topic);
         if (null == topicPublishInfo || !topicPublishInfo.ok()) {
             this.topicPublishInfoTable.putIfAbsent(topic, new TopicPublishInfo());
-            this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic);
             topicPublishInfo = this.topicPublishInfoTable.get(topic);
+            if (!topicPublishInfo.checkNotFoundFlag()) {
```

Comment:
Is topicPublishInfo possible to be null

Reply:
(line715)
`this.topicPublishInfoTable.putIfAbsent(topic, new TopicPublishInfo());`
So theoretically topicPublishInfo is not empty.